1-org/modules/centralized-logging/main.tf

```diff
@@ -0,0 +1,204 @@
+}
+
+resource "random_string" "suffix" {
+  length  = 4
+  upper   = false
+  special = false
+}
+
+module "log_export" {
+  source  = "terraform-google-modules/log-export/google"
+  version = "~> 7.3.0"
+
+  for_each = local.all_sinks
+
+  destination_uri        = lookup(each.value, "destination_uri", lookup(local.destination_uri_map, substr(each.key, -3, -1), ""))
+  filter                 = lookup(each.value, "logging_sink_filter", "")
+  log_sink_name          = lookup(each.value, "logging_sink_name", "sk-to-${lookup(local.logging_sink_name_map, substr(each.key, -3, -1), "log_dest_")}")
+  parent_resource_id     = substr(each.key, 0, length(each.key) - 4)
+  parent_resource_type   = var.resource_type
+  unique_writer_identity = true
+  include_children       = tobool(lookup(each.value, "include_children", "false"))
+  bigquery_options       = substr(each.key, -3, -1) == "bgq" ? local.bgq_options_part_tables : null
+}
+
+#-------------------------#
+# Send logs to Log Bucket #
+#-------------------------#
+module "destination_logbucket" {
+  //  source  = "terraform-google-modules/log-export/google//modules/logbucket"
+  //  version = "~> 7.4.2"
+
+  source = "github.com/terraform-google-modules/terraform-google-log-export//modules/logbucket"
+
+  count = var.logbucket_options != null ? 1 : 0
+
+  project_id                    = var.logging_destination_project_id
+  name                          = lookup(var.logbucket_options, "name", "${lookup(local.logging_target_name_prefix, "lbk", "log_dest_")}${random_string.suffix.result}")
+  log_sink_writer_identity      = module.log_export["${local.value_first_resource}_lbk"].writer_identity
+  location                      = lookup(var.logbucket_options, "location", "global")
+  retention_days                = lookup(var.logbucket_options, "retention_days", 30)
+  grant_write_permission_on_bkt = false
+}
+
+#-------------------------------------------#
+# Log Bucket Service account IAM membership #
+#-------------------------------------------#
+resource "google_project_iam_member" "logbucket_sink_member" {
+  for_each = var.logbucket_options != null ? local.logbucket_sink_member : {}
+
+  project = var.logging_destination_project_id
+  role    = "roles/logging.bucketWriter"
+  member  = module.log_export["${each.value}_lbk"].writer_identity
+}
+
+
+#-----------------------#
+# Send logs to BigQuery #
+#-----------------------#
+module "destination_bigquery" {
+  source  = "terraform-google-modules/log-export/google//modules/bigquery"
+  version = "~> 7.3.0"
+
+  count = var.bigquery_options != null ? 1 : 0
+
+  project_id                 = var.logging_destination_project_id
+  dataset_name               = replace(lookup(var.bigquery_options, "dataset_name", "${lookup(local.logging_target_name_prefix, "bgq", "log_dest_")}${random_string.suffix.result}"), "-", "_")
```

Comment:
Can we move default names to a local instead for readability.

Reply:
Done